### PR TITLE
Two changes to thrift-logger module

### PR DESCRIPTION
**Pull request by zzhhhzz**
1. upgrade logback to 1.1.8
2. configure thread to sleep x seconds before closing thrift loggers.

### CODE REVIEW

File: thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java [hunk 1]
```diff
@@ -34,6 +35,16 @@
   protected ConcurrentHashMap<String, ThriftLogger> loggersByTopic
       = new ConcurrentHashMap<String, ThriftLogger>();
 
+  private int sleepInSecBeforeCloseLoggers = -1;
```
Comment (vahidhashemian) [inline thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java]: It seems this number gets converted to ms value later. Why not define this as `sleepInMsBeforeCloseLoggers`?

File: thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java [hunk 2]
```diff
@@ -93,8 +104,19 @@ private synchronized ThriftLogger getOrCreateLogger(ThriftLoggerConfig thriftLog
...
   public synchronized void shutdown() {
-    for (ThriftLogger logger : loggersByTopic.values()) {
-      logger.close();
+    try{
+      if (sleepInSecBeforeCloseLoggers > 0){
+        LOGGER.info("Before closing loggers, sleep {} seconds", sleepInSecBeforeCloseLoggers);
+        Thread.sleep(sleepInSecBeforeCloseLoggers * 1000);
+        LOGGER.info("After {} seconds, start to close loggers", sleepInSecBeforeCloseLoggers);
+      }
+    }catch(InterruptedException e){
```
Comment (HenryCaiHaiying) [inline thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java]: missing space

File: thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java [hunk 3]
```diff
@@ -93,8 +104,19 @@ private synchronized ThriftLogger getOrCreateLogger(ThriftLoggerConfig thriftLog
...
+        LOGGER.warn("Thread got interrupted", e);
+    } finally{
```
Comment (HenryCaiHaiying) [inline thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java]: missing space

File: thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java (hunk 1 above)
Comment (zzhhhzz) [reply thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java]: In singer-common module ( https://github.com/pinterest/singer/blob/master/singer-commons/src/main/java/com/pinterest/singer/loggingaudit/client/AuditEventKafkaSender.java#L82 ),  we have similar parameter (in seconds) to control the shutdown behavior.   I feel it is better to be consistent in naming across modules.

File: thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java (hunk 2 above)
Comment (zzhhhzz) [reply thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java]: fix in the update of the PR

File: thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java (hunk 3 above)
Comment (zzhhhzz) [reply thrift-logger/src/main/java/com/pinterest/singer/client/BaseThriftLoggerFactory.java]: fix in the update of the PR

**Pull request merged**